### PR TITLE
snakemake: Updated to new version 7.0.2

### PR DESCRIPTION
**Pull request by marcusboden**
Added new snakemake versions and a dependency

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-yte/package.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    structure in combination with Python expressions for enabling to
+    dynamically build YAML documents."""
+
+    homepage = "https://github.com/koesterlab/yte"
+    pypi = "yte/yte-1.2.0.tar.gz"
+    maintainers = ['marcusboden']
+    version('1.2.0', '0368f220bb96fb3290bbd5a90e3d218af483d0d3e5abf9b08b24e29b150f151e')
+
+    depends_on('python@3.7:', type=('build', 'run'))
+    depends_on('py-poetry', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yte/package.py]:
```suggestion
    depends_on('py-poetry-core@1:', type='build')
```

File: var/spack/repos/builtin/packages/snakemake/package.py [hunk 2]
```diff
@@ -24,6 +26,11 @@ class Snakemake(PythonPackage):
 
     depends_on('python@3.3:3.6', when='@:5')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/snakemake/package.py]: Looks like 3.7+ is required now

File: var/spack/repos/builtin/packages/snakemake/package.py [hunk 3]
```diff
@@ -24,6 +26,11 @@ class Snakemake(PythonPackage):
...
 
+    with when('@7:'):
+        depends_on('py-yte', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/snakemake/package.py]:
```suggestion
        depends_on('py-yte@1', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/snakemake/package.py [hunk 4]
```diff
@@ -24,6 +26,11 @@ class Snakemake(PythonPackage):
...
+        depends_on('py-jinja2', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/snakemake/package.py]:
```suggestion
        depends_on('py-jinja2@3', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/snakemake/package.py (hunk 2 above)
Comment (marcusboden) [reply var/spack/repos/builtin/packages/snakemake/package.py]: Ah right. I'll add that in the '@7' part

File: var/spack/repos/builtin/packages/snakemake/package.py [hunk 5]
```diff
@@ -24,6 +26,12 @@ class Snakemake(PythonPackage):
 
     depends_on('python@3.3:3.6', when='@:5')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/snakemake/package.py]: Both of these should presumably be `type=('build', 'run')`? Unless it also links or Python isn't needed at run-time.

Comment (marcusboden) [reply var/spack/repos/builtin/packages/snakemake/package.py]: Yes, you're right. Fixed it.

**Review by adamjstewart — DISMISSED** [review]

**Comment by adamjstewart** [comment]
You'll need to add a new version of py-plac to get the tests to pass.

**Comment by marcusboden** [comment]
I've added the new py-plac version. And also the most recent Snakemake version, since they had some updates since I made the initial PR.

**Review by adamjstewart — CHANGES_REQUESTED** [review]
filelock is no longer needed in the latest version

**Comment by adamjstewart** [comment]
Ping @marcusboden

**Comment by marcusboden** [comment]
Hi, sorry for the delay. I've switched jobs and no longer have access to the repo I made the PR from, so I cannot make any changes to it anymore.

If you can, just add `when='@6'` to py-filelock. Or we can just close the PR.

**Comment by adamjstewart** [comment]
#34041 adds an even newer version, let's do that instead.

**Pull request closed without merging**